plugin/pkg/utils/cloudprofile.go

```diff
@@ -60,13 +59,16 @@ func ValidateCloudProfileChanges(namespacedCloudProfileLister gardencorev1beta1l
 	if err != nil {
 		return err
 	}
-	if oldCloudProfileReference.Kind == constants.CloudProfileReferenceKindCloudProfile &&
-		newCloudProfileReference.Kind == constants.CloudProfileReferenceKindNamespacedCloudProfile &&
-		equality.Semantic.DeepEqual(oldCloudProfileReference, newCloudProfileRoot) {
+	oldCloudProfileRoot, err := getRootCloudProfile(namespacedCloudProfileLister, oldCloudProfileReference, oldShoot.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if equality.Semantic.DeepEqual(oldCloudProfileRoot, newCloudProfileRoot) {
 		return nil
 	}
 
-	return fmt.Errorf("a Seed's CloudProfile may only be changed to a descendant NamespacedCloudProfile, other modifications are currently not supported")
+	return fmt.Errorf("cloud profile reference change is invalid: cannot change from %q (root: %q) to %q (root: %q). The cloud profile reference must remain within the same hierarchy", oldCloudProfileReference.Name, oldCloudProfileRoot.Name, newCloudProfileReference.Name, newCloudProfileRoot.Name)
```

Comment:
This message seems confusing if one tries to change from a `CloudProfile` a -> `CloudProfile` b. Does it make sense to handle this case separately?

Reply:
Please recheck, whether you like the [new suggestion](https://github.com/gardener/gardener/pull/10811/commits/4ee212a0e7ae398b54e6428a631bba1088ca37f0#diff-02239b80aadad29ea3a581b905127dc9d89bf96dd9f2a27c6d43aa5dd57dd9e4R70-R79) better.